server/sse.go

```
@@ -356,31 +356,37 @@ func (s *SSEServer) handleMessage(w http.ResponseWriter, r *http.Request) {
-			// Session is closed, don't try to queue
-		default:
-			// Queue is full, could log this
-		}
+		// Only send response if there is one (not for notifications)
+		if response != nil {
+			var message string
+			eventData, err := json.Marshal(response)
+			if err != nil {
+				// If there is an error marshalling the response, send a generic error response
+				log.Printf("failed to marshal response: %v", err)
+				message = fmt.Sprintf("event: message\ndata: {\"error\": \"internal error\",\"jsonrpc\": \"2.0\", \"id\": null}\n\n")
```

golangci-lint (1.64.8)

S1039: unnecessary use of fmt.Sprintf

(gosimple)